editor/src/messages/portfolio/document/node_graph/node_graph_message_handler.rs

```diff
@@ -381,8 +381,23 @@ impl NodeGraphMessageHandler {
 			warn!("Deleting the output node!");
 			return false;
 		}
+
+		let mut first_input_node: Option<NodeInput> = None;
+		let mut first_output_node: Option<&NodeOutput> = None;
+
 		for (node_id, node) in network.nodes.iter_mut() {
 			if *node_id == deleting_node_id {
+				if node.inputs.len() > 0 {
+					first_input_node = Some(node.inputs[0].clone());
+					warn!("first node: {:?}", first_input_node);
+
+					// fetch output node only if there's an input node for comparison too
+					if network.outputs.len() > 0 {
+						first_output_node = Some(&network.outputs[0]);
+						warn!("first output node: {:?}", first_output_node);
+					}
+				}
+
```

Comment:
There is no guarantee that the deleted node will appear in the hashmap before the other nodes. Hashmaps by definition do not guarantee ordering.

Instead you must find the deleted_node before this loop, and set the first_input_node variable based on this.

You must also ensure that the input actually refers to a node i.e. `if let NodeInput::Node { node_id, .. } = input`.